daemon/system-info.sh

```diff
@@ -430,10 +430,24 @@ fi
 # Detect instance metadata for VMs running on cloud providers
 
 CLOUD_TYPE="unknown"
+CLOUD_PLATFORM="unknown"
 CLOUD_INSTANCE_TYPE="unknown"
 CLOUD_INSTANCE_REGION="unknown"
-
-if [ "${VIRTUALIZATION}" != "none" ] && command -v curl > /dev/null 2>&1; then
+CLOUD_INSTANCE_ZONE="unknown"
+CLOUD_DETECTION="none"
+
+if command -v cloud-init > /dev/null 2>&1 && cloud-init query --all > /dev/null 2>&1; then
+  CLOUD_TYPE="$(cloud-init query v1.cloud_name)"
+  CLOUD_PLATFORM="$(cloud-init query v1.platform)"
+  CLOUD_INSTANCE_REGION="$(cloud-init query v1.region)"
+  CLOUD_INSTANCE_ZONE="$(cloud-init query v1.availability_zone)"
```

Comment:
Doesn't look like this gives meaningful result on DigitalOcean:

```cmd
ilyam@deb-fra1-01:~$ cloud-init query v1.cloud_name
configdrive

ilyam@deb-fra1-01:~$ cloud-init query v1.platform
configdrive

ilyam@deb-fra1-01:~$ cloud-init query v1.region
null

ilyam@deb-fra1-01:~$ cloud-init query v1.availability_zone
fra1
```

Reply:
Hmm, looks like the cloud-init documentation is not exactly accurate when it comes to the values for the platform and cloud_name keys. They _should_ be reporting based on the actual hosting, but they appear to actually be reporting info based on what datasource plugin it’s using.

That said, _something_ appears to be going wrong there otherwise, because there is a specific DigitalOcean datasource that pulls from their IMDS, and it should be positively ID'ing DO and using that datasource.